src/future_jiffies_patch.c

```diff
@@ -53,11 +59,26 @@ static int __init custom_init(void)
 		return -1;
 	}
 
+	/*
+	 * minor:
+	 * this case can be done even without fops using:
+	 * debugfs_create_u64(..., (u64 *)&jiffies);
+	 */
 	file1 = debugfs_create_file("jiffies", 0004, dir, NULL, &fops_j);
 
 	if (!file1) {
 		// Abort module load.
 		pr_alert("debugfs_kernelcare: failed to create /sys/kernel/debug/kernelcare/jiffies\n");
+		/*
+		 * cleanup, right?
+		 * otherwise whichever debugfs_create_dir() did will not be reverted, see
+		 * https://elixir.bootlin.com/linux/latest/source/fs/debugfs/inode.c#L581
+		 * PS: in the following version this issue is fixed - ok
+		 *
+		 * Also, such functions have a tricky return values in case of errors.
+		 * like ERR_PTR(-ERROR)
```

Comment:
incorrect error checking, don't worry, just read the docs/Linux sources carefully

Reply:
I would fix it like that
```
static int __init custom_init(void)
{
	struct dentry *file1;

	dir = 0;
	dir = debugfs_create_dir("kernelcare", 0);
	if (IS_ERR(dir)) {
		// Abort module load.
		pr_alert("debugfs_kernelcare: failed to create /sys/kernel/debug/kernelcare\n");
		return -1;
	}

	file1 = debugfs_create_file("jiffies", 0004, dir, NULL, &fops_j);

	if (IS_ERR(file1)) {
		// Abort module load.
		pr_alert("debugfs_kernelcare: failed to create /sys/kernel/debug/kernelcare/jiffies\n");
		return -1;
	}

	 pr_debug("Hello KernelCare!");
	return 0;
}
```